lib/lockbox/model.rb

```diff
@@ -137,21 +137,22 @@ def attributes
                 # essentially a no-op if already loaded
                 # an exception is thrown if decryption fails
                 self.class.lockbox_attributes.each do |_, lockbox_attribute|
-                  # don't try to decrypt if no decryption key given
-                  next if lockbox_attribute[:algorithm] == "hybrid" && lockbox_attribute[:decryption_key].nil?
-
                   # it is possible that the encrypted attribute is not loaded, eg.
                   # if the record was fetched partially (`User.select(:id).first`).
                   # accessing a not loaded attribute raises an `ActiveModel::MissingAttributeError`.
-                  encrypted_attribute = lockbox_attribute[:encrypted_attribute]
-                  if has_attribute?(encrypted_attribute)
-                    send(lockbox_attribute[:attribute])
+                  if has_attribute?(lockbox_attribute[:encrypted_attribute])
+                    begin
+                      send(lockbox_attribute[:attribute])
+                    rescue ArgumentError => e
+                      raise e if e.message != "No decryption key set"
+                    end
                     next
```

Comment:
one question: even though the upstream added the above check:

```
if has_attribute?(lockbox_attribute[:encrypted_attribute])
   ...
   send(lockbox_attribute[:attribute])
   ...
end
```

They did _*not*_ add a `next` statement ~~ was wondering why we added a `next` statement here in this case? @RickCSong @nealharris can you shed some light since ya'll did the original commit?

Reply:
Hmm, I'm not sure offhand, but it might have been to match what's going on in https://github.com/persona-id/persona-web/blob/master/config/initializers/ext/store_model/encrypted.rb